tests/integration/purge_caches.rs

```diff
@@ -0,0 +1,103 @@
+use failure::Error;
+use rustwide::cmd::SandboxBuilder;
+use rustwide::{Crate, Toolchain};
+use sha1::{Digest, Sha1};
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+const WORKSPACE_NAME: &str = "purge-caches";
+
+#[test]
+fn test_purge_caches() -> Result<(), Error> {
+    let workspace_path = crate::utils::workspace_path(WORKSPACE_NAME);
+    let workspace = crate::utils::init_named_workspace(WORKSPACE_NAME)?;
+
+    // Do an initial purge to prevent stale files from being present.
+    workspace.purge_all_build_dirs()?;
+    workspace.purge_all_caches()?;
+
+    let toolchain = Toolchain::dist("stable");
+    toolchain.install(&workspace)?;
+
+    let start_contents = WorkspaceContents::collect(&workspace_path)?;
+
+    let crates = vec![
+        Crate::crates_io("lazy_static", "1.0.0"),
+        Crate::git("https://github.com/pietroalbini/git-credential-null"),
+    ];
+
+    // Simulate a build, which is going to fill up the caches.
+    for krate in &crates {
+        krate.fetch(&workspace)?;
```

Comment:
Could this cause an index update if it has changed between the init and now?

Reply:
No, rustwide doesn't use `cargo` to download crates or git repositories.

Reply:
Whelp there was a race condition anyway.

Reply:
Ah, yes, but what about the `check` call during the build?

```
file .workspaces/purge-caches/cargo-home/registry/index/github.com-1ecc6299db9ec823/.git/FETCH_HEAD changed
```

https://github.com/rust-lang/rustwide/pull/26/checks?check_run_id=735677032#step:6:528

Reply:
(Though I must still be missing something cause it looks like the registry should be readonly during that).

Reply:
The index is updated while generating the lockfile for `lazy_static`. I'll address this tomorrow.